core/src/main/java/org/elasticsearch/index/mapper/GeoShapeFieldMapper.java

```diff
@@ -179,6 +196,9 @@ public GeoShapeFieldMapper build(BuilderContext context) {
                 } else if (Names.STRATEGY.equals(fieldName)) {
                     builder.fieldType().setStrategyName(fieldNode.toString());
                     iterator.remove();
+                } else if (fieldName.equals("ignore_malformed")) {
+                    builder.ignoreMalformed(TypeParsers.nodeBooleanValue(fieldName, "ignore_malformed", fieldNode, parserContext));
```

Comment:
nit: maybe the field name could be a constant under `Names`, the same as in e.g. GeoPointFieldMapper (or reuse that one)